### PR TITLE
treewide: Remove eSPI Slave from Kconfig and eSPI API

### PR DESCRIPTION
**Pull request by albertofloyd**
Adjust terms per eSPI specification 1.5 
* Replace CONFIG_ESPI_SLAVE by CONFIG_ESPI_TARGET
* Replace CONFIG_ESPI_SAF by CONFIG_ESPI_TAF
* Replace ESPI_BUS_SAF_NOTIFICATION with ESPI_TAF_BUS_NOTIFICATION in API

### CODE REVIEW

File: doc/releases/migration-guide-3.7.rst [hunk 1]
```diff
@@ -342,6 +342,8 @@ Enhanced Serial Peripheral Interface (eSPI)
   ``ESPI_VWIRE_SIGNAL_TARGET_BOOT_STS``, ``ESPI_VWIRE_SIGNAL_TARGET_BOOT_DONE`` and
   ``ESPI_VWIRE_SIGNAL_TARGET_GPIO_<NUMBER>`` respectively to reflect the new terminology
   in eSPI 1.5 specification. (:github:`68492`)
+  The KConfig ``CONFIG_ESPI_SLAVE`` was renamed to ``CONFIG_ESPI_TARGET``, similarly
+  ``CONFIG_ESPI_SAF`` was renamed as ``CONFIG_ESPI_TAF``
```
Comment (aescolar) [inline doc/releases/migration-guide-3.7.rst:346]:
```suggestion
  ``CONFIG_ESPI_SAF`` was renamed as ``CONFIG_ESPI_TAF`` (:github:`73887`)
```

File: drivers/espi/CMakeLists.txt [hunk 2]
```diff
@@ -10,8 +10,8 @@ zephyr_library_sources_ifdef(CONFIG_ESPI_NPCX		host_subs_npcx.c)
 zephyr_library_sources_ifdef(CONFIG_ESPI_TAF_NPCX	espi_taf_npcx.c)
 zephyr_library_sources_ifdef(CONFIG_USERSPACE		espi_handlers.c)
 zephyr_library_sources_ifdef(CONFIG_ESPI_EMUL		espi_emul.c)
-zephyr_library_sources_ifdef(CONFIG_ESPI_SAF_XEC	espi_saf_mchp_xec.c)
```
Comment (scottwcpg) [inline drivers/espi/CMakeLists.txt:13]: Is is possible to change the source file name from espi_saf_mchp_xec.c to espi_taf_mchp_xec.c
Same for espi_saf_mchp_xec_v2.c

Comment (keith-zephyr) [reply drivers/espi/CMakeLists.txt:13]: That can be done as a separate commit or separate PR.

**Review by keith-zephyr — DISMISSED** [review]

File: drivers/espi/CMakeLists.txt (hunk 2 above)
Comment (scottwcpg) [reply drivers/espi/CMakeLists.txt:13]: ok

**Review by scottwcpg — DISMISSED** [review]

File: doc/releases/migration-guide-3.7.rst (hunk 1 above)
Comment (albertofloyd) [reply doc/releases/migration-guide-3.7.rst]: Done

**Review by keith-zephyr — APPROVED** [review]

**Review by scottwcpg — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**